docs/superpowers/specs/2026-04-24-custom-fields-design.md

```
@@ -0,0 +1,508 @@
+    public function serialize(CustomField $field, mixed $input): ?string;   // UI input  → TEXT column
+    public function deserialize(CustomField $field, ?string $stored): mixed; // TEXT column → typed PHP
+    public function formatForDisplay(CustomField $field, ?string $stored): string; // TEXT → rendered
+}
+```
+
+Storage format in the single `value TEXT` column:
+
+| Type | Stored as | Deserialized PHP type |
+|---|---|---|
+| TEXT / TEXTAREA / EMAIL / URL | string as-is | `string` |
+| NUMBER | numeric string (preserves precision) | `int|float` (on read) |
```

markdownlint-cli2 (0.22.1)

[warning] Table column count
Expected: 3; Actual: 4; Too many cells, extra data will be missing

(MD056, table-column-count)